web/js/mapUI/components/layers/addLayer.js

```diff
@@ -51,7 +51,12 @@ function AddLayer(props) {
       compareMapUi.create(ui.selected, mode);
     } else {
       const newLayer = await createLayer(def);
-      ui.selected.getLayers().insertAt(index, newLayer);
+      const layers = ui.selected.getLayers();
+      if (index <= layers.getLength()) {
```

Comment:
This feels like a bit of a hacky fix to me. Why is this behavior different than when adding any other layers? Surely there is a solution to this further upstream that solves the actual problem, rather than accounting for the problem later on.

Reply:
I'm not sure fixing the source of the bug is feasible within the scope of this story. The source of the issue is a race condition caused by [this useEffect](https://github.com/nasa-gibs/worldview/blob/main/web/js/mapUI/components/update-projection/updateProjection.js#L367) hook which is unfortunately critical to how AERONET layers are rendered. Modifying said hook to only reload layers when the newly rendered palette is from AERONET works for the most part but unfortunately causes AERONET layers to not render correctly when hydrated from the url.